services/meta/client.go

```diff
@@ -676,15 +676,15 @@ func (c *Client) PruneShardGroups() error {
 	data := c.cacheData.Clone()
 	for i, d := range data.Databases {
 		for j, rp := range d.RetentionPolicies {
+			remainingshards := data.Databases[i].RetentionPolicies[j].ShardGroups[:0]
```

Comment:
I should definitely change the name of the variable. I think ```remainingShardGroups``` would be even better. But if I declare a new variable, we will loose the "no allocation" we gain using this trick. In the end, we are not allocating a new array, just using the old one. After the appends, it will have the size that we need, but, if fact, what we did was change the contents of the original array.
But it is up to you to decide... Maybe we can consider a premature optimization...